### PR TITLE
Save playlist

### PR DESCRIPTION
**Pull request by olegrise**
## Description
- Save playlist
- Load playlist
- Load templates name and status

## Motivation and Context
Epic requirements

## How Has This Been Tested?
Will be added later

## Release Plan:
- As the Submitter, upon requesting review of this pull request, I confirm that the [Release Checklist](https://help.risevision.com/hc/en-us/articles/360031119991) has been completed. 
- As the Reviewer, upon approving the changes in this PR, I confirm I have reviewed and I agree that the [Release Checklist](https://help.risevision.com/hc/en-us/articles/360031119991) has been completed

#### Release Checklist Items Skipped?
If any Release Checklist items were intentionally skipped, please provide which ones and the reasons why

### CODE REVIEW

File: web/scripts/template-editor/components/directives/dtv-component-playlist.js [hunk 1]
```diff
@@ -43,6 +46,53 @@ angular.module('risevision.template-editor.directives')
...
+          $scope.jsonToSelectedTemplates = function (json) {
+            var result = [];
+            var parsedItems;
+
+            if (json) {
+              parsedItems = JSON.parse(json);
+            }
+
+            if (Array.isArray(parsedItems)) {
+              _.forEach(parsedItems, function (item) {
```
Comment (andrecardoso) [inline web/scripts/template-editor/components/directives/dtv-component-playlist.js]: I think this is  a good scenario for `_.map` as you are transforming the entire array, something like:
```javacript
         result = _.map(parsedItems, function (item) {
                  var playlistItem = {
                    'duration': item.duration,
                    'play-until-done': item['play-until-done'],
                    'transition-type': item['transition-type'],
                    'id': item.element && item.element.attributes ? item.element.attributes['presentation-id'] : undefined,
                    'productCode': item.element && item.element.attributes ? item.element.attributes['template-id'] : undefined
                  };
                  return playlistItem;
              });
```

File: web/scripts/template-editor/components/directives/dtv-component-playlist.js [hunk 2]
```diff
@@ -43,6 +46,53 @@ angular.module('risevision.template-editor.directives')
...
+  
+                result.push(playlistItem);
+              });
+            }
+
+            return result;
+          };
+
+          $scope.selectedTemplatesToJson = function () {
+            var playlistItems = [];
```
Comment (andrecardoso) [inline web/scripts/template-editor/components/directives/dtv-component-playlist.js]: Same as above, `_.map` would fit like a glove

File: web/scripts/template-editor/components/directives/dtv-component-playlist.js [hunk 3]
```diff
@@ -53,6 +103,35 @@ angular.module('risevision.template-editor.directives')
...
+            _.forEach(templates, function (item) {
+              presentationIds.push('id:' + item.id);
+            });
+
+            var search = {filter: presentationIds.join(' OR ')};
+
+            presentation.list(search)
+            .then(function(res) {
+              if (res.items) {
+                _.forEach(templates, function (template) {
```
Comment (andrecardoso) [inline web/scripts/template-editor/components/directives/dtv-component-playlist.js]: I think you can use `_.findWhere` to make this more readable:

```javascript

_.forEach(templates, function (template) {
    var item = _.findWhere(res.items, {id: tempalte.id});
    if (item) { // maybe not required if res.items will always contain the template
        template.name = item.name;
        template.revisionStatusName = item.revisionStatusName;
    }
});
```

Comment (olegrise) [reply web/scripts/template-editor/components/directives/dtv-component-playlist.js]: @andrecardoso I could figure out how to use `_.findWhere`, besides it looks like this method is removed in the latest version of Lodash. I don't think it matters much in this case. Thanks.

Comment (andrecardoso) [reply web/scripts/template-editor/components/directives/dtv-component-playlist.js]: No problem. Thanks

**Pull request merged**